### DISCUSSION TITLE
Re: TFT_Meter - How to draw two Needles on the same tft object?

### DISCUSSION THREAD

**Discussion by mprowe** [Bodmer/TFT_eSPI#1765] [Q&A - General] [answered]
I would like to have two meters (say Volts & Amps) displayed on the same display.
Starting with your example TFT_Meter, I have been able to deal with the two faces (the second offset by Y).
However, my efforts to create two instances plotNeedle() have failed.

I have seen the question  #1415 and your pointers to the two examples. Unfortunately, it it obvious that I do not understand what is involved! I have attached my proto Needle class files and a section of my mod'ed TFT_Meter.ino in the hope that it explains what I am trying to achieve.

![TFT_Meter ino](https://user-images.githubusercontent.com/14279344/162609059-fbbf971c-09c9-458c-a09d-c2e8fc9d607b.png)
[myMeter.zip](https://github.com/Bodmer/TFT_eSPI/files/8458558/myMeter.zip)

Any tips/pointer will be gratefully received. Regards, M.

**Comment by Bodmer** [comment, marked as answer]
The myMeter example does not look like it plots two meters, but here are a few comments:

1. An address is being sent to the constructor so it has to accept a pointer to the TFT_eSPI functions:
In Needle.h:
```
Needle(TFT_eSPI* tft);
```
In Needle.cpp:
```
Needle::Needle(TFT_eSPI* tft)
```
2. A class variable will be needed to hold the pointer:
```
TFT_eSPI* ntft;
```
3. The processor may crash if you try to use Serial.print in the constructor, because the Serial class has not been loaded yet when the Needle instances are created.
4. To call the TFT_eSPI library functions from the class you would need to use the funmction pointer with "->" instead of "."  thus:
```
ntft->fillScreen(TFT_RED);
```
If you work on the sketch some more and get stuck then post back and I will help some more.

**Reply by Bodmer** [reply]
I have updated a copy for you:
[myMeter.zip](https://github.com/Bodmer/TFT_eSPI/files/8459388/myMeter.zip)

**Reply by mprowe** [reply]
Thank you for you help Bodmar,
This is what I've ended up with. Which would not have been possible without your foundation work.
![myMeter](https://user-images.githubusercontent.com/14279344/162981964-a9372e7b-b6d8-4087-a60b-49fce408b449.png)
[myMeter.zip](https://github.com/Bodmer/TFT_eSPI/files/8473646/myMeter.zip)
Very many thanks, M.

**Reply by Bodmer** [reply]
Thanks for sharing. Glad to see you have your project up and running.
